bsm_generator/src/bsm_generator_node.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (C) 2019-2022 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include "bsm_generator/bsm_generator_node.hpp"
+
+namespace bsm_generator
+{
+  namespace std_ph = std::placeholders;
+
+  BSMGenerator::BSMGenerator(const rclcpp::NodeOptions &options)
+      : carma_ros2_utils::CarmaLifecycleNode(options)
+  {
+    // Create initial config
+    config_ = Config();
+
+    // Declare parameters
+    config_.bsm_generation_frequency = declare_parameter<double>("bsm_generation_frequency", config_.bsm_generation_frequency);
+    config_.bsm_id_rotation_enabled  = declare_parameter<bool>("bsm_id_rotation_enabled", config_.bsm_id_rotation_enabled);
+    config_.bsm_message_id           = declare_parameter<int>("bsm_message_id", config_.bsm_message_id);
+    config_.vehicle_length           = declare_parameter<double>("vehicle_length", config_.vehicle_length);
+    config_.vehicle_width            = declare_parameter<double>("vehicle_width", config_.vehicle_width);
+  }
+
+  rcl_interfaces::msg::SetParametersResult BSMGenerator::parameter_update_callback(const std::vector<rclcpp::Parameter> &parameters)
+  {
+    auto error   = update_params<double>({{"bsm_generation_frequency", config_.bsm_generation_frequency}}, parameters);
+    auto error_2 = update_params<bool>({{"bsm_id_rotation_enabled", config_.bsm_id_rotation_enabled}}, parameters);
+    auto error_3 = update_params<int>({{"bsm_message_id", config_.bsm_message_id}}, parameters);
+    auto error_4 = update_params<double>({{"vehicle_length", config_.vehicle_length}}, parameters);
+    auto error_5 = update_params<double>({{"vehicle_width", config_.vehicle_width}}, parameters);
```

Comment:
You can combine all the calls of the same type 
```
auto error = update_params<double>({
    {"bsm_generation_frequency", config_.bsm_generation_frequency},
    {"vehicle_length", config_.vehicle_length},
    {"vehicle_width", config_.vehicle_width}
}, parameters);
```
Just makes it a bit cleaner to read since you'll get 3 update params calls (bool, int, double) instead of 5

Reply:
Oh interesting, I didn't know these could be combined.  Updated.